desktop/src/gui/controller.rs

```diff
@@ -286,3 +289,49 @@ impl GuiController {
         self.gui.open_file_advanced()
     }
 }
+
+// try to load known unicode supporting fonts to draw cjk characters in egui
+fn load_system_fonts() -> anyhow::Result<egui::FontDefinitions> {
+    let mut font_database = Database::default();
+    font_database.load_system_fonts();
+
+    let system_unicode_fonts = Query {
+        families: &[
+            Family::Name("MS UI Gothic"),     // windows
+            Family::Name("Arial Unicode MS"), // macos
+            Family::Name("Noto Sans"),        // linux
+            Family::SansSerif,
+        ],
+        ..Query::default()
+    };
+
+    let id = font_database
+        .query(&system_unicode_fonts)
+        .ok_or(anyhow!("no unicode fonts found!"))?;
```

Comment:
I think it'd be better to log an error and continue, rather than panic. Unless including `SansSerif` is supposed to make it infallible?

Reply:
ok_or doesn't panic, I return immediately and load the default fonts if anything goes wrong- I should log it though, right now it does silently fail